### PR TITLE
Rebuild Pikelet using normalization by evaluation 

### LINKED ISSUE #14: Error recovery in the type checker
We should be able to run programs that:

- are broken syntactically
- are partially type-incorrect

We should also be able to see more than one type error message.

At the moment we recover from syntactic errors, inserting `concrete::Term::Error` variants in the places that we fail to parse. But then we blow up if wen try to desugar these into `RawTerm`s. Instead we should probably insert a `static-error : {a : Type} -> String -> a` function in those places instead...

### PR DESCRIPTION
**Pull request by brendanzab**
See issue #196 for more information!

Closes #14, #62, #109, #181, #196.

### CODE REVIEW

**Comment by brendanzab** [comment]
Note that this is such a big change that I don't really expect anyone to read the diff. It might be better to just have a look at the source directly.

File: pikelet/Cargo.toml [hunk 1]
```diff
@@ -21,7 +21,7 @@ itertools = "0.8"
 lalrpop-util = "0.17"
 pretty = "0.9"
 regex = "1.3"
-logos = "0.9.7"
+logos = " 0.11.0-rc2"
```
Comment (bjorn3) [inline pikelet/Cargo.toml:24]:
```suggestion
logos = "0.11.0-rc2"
```

Comment (brendanzab) [reply pikelet/Cargo.toml:24]: Oh, woops! Thanks for catching this!

**Comment by brendanzab** [comment]
This doesn't yet complete all of the stuff in #196, but I'm thinking I'm going to go ahead an merge this!

**Pull request merged**